studio/schemas/objects/eventDate.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import type { Rule, ValidationContext } from '@sanity/types'
+import TimeInput, { EMPTY, INVALID_TIME_FORMAT } from '../components/TimeInput'
+
+export type EventDate = {
+  _type: 'eventDate'
+  date?: string
+  startTime?: string
+  endTime?: string
+}
+
+const isValid = (field: string | undefined) => {
+  const time = field?.split(':') ?? [EMPTY, EMPTY]
+  return !(time.includes(EMPTY) && time[0] !== time[1])
+}
+
+export default {
+  title: 'Fields for event date',
+  name: 'eventDate',
+  type: 'object',
+  fields: [
+    {
+      name: 'date',
```

Comment:
I would add a short description where we also state the date format, just so it's clear at first glance